paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -49,518 +46,589 @@
 log.addHandler(logging.NullHandler())
 
 
-def get_cluster_metrics_provider(name):
-    """
-    Returns a cluster metrics provider matching the given name.
-    """
-    return _autoscaling_components[CLUSTER_METRICS_PROVIDER_KEY][name]
+class ClusterAutoscaler(object):
 
+    def __init__(self, resource, pool_settings, config_folder, dry_run):
+        self.resource = resource
+        self.pool_settings = pool_settings
+        self.config_folder = config_folder
+        self.dry_run = dry_run
 
-def get_scaler(name):
-    """
-    Returns a scaler matching the given name.
-    """
-    return _autoscaling_components[SCALER_KEY][name]
+    def set_capacity(self, capacity):
+        pass
 
+    def get_instance_type_weights(self):
+        pass
 
-def get_spot_fleet_instances(spotfleet_request_id, region=None):
-    ec2_client = boto3.client('ec2', region_name=region)
-    spot_fleet_instances = ec2_client.describe_spot_fleet_instances(
-        SpotFleetRequestId=spotfleet_request_id)['ActiveInstances']
-    return spot_fleet_instances
+    def describe_instances(self, instance_ids, region=None, instance_filters=None):
+        """This wraps ec2.describe_instances and catches instance not
+        found errors. It returns a list of instance description
+        dictionaries. It assumes one instance per reservation (which
```

Comment:
Ah, I've actually fixed this because it doesn't hold for ASGs. In fact I could do with updating a few docstrings! 😢